adafruit_rfm69.py

```diff
@@ -714,118 +753,172 @@ def send(
         # pylint: disable=len-as-condition
         assert 0 < len(data) <= 60
-        assert len(tx_header) == 4, "tx header must be 4-tuple (To,From,ID,Flags)"
         # pylint: enable=len-as-condition
         self.idle()  # Stop receiving to clear FIFO and keep it clear.
         # Fill the FIFO with a packet to send.
-        with self._device as device:
-            self._BUFFER[0] = _REG_FIFO | 0x80  # Set top bit to 1 to
-            # indicate a write.
-            self._BUFFER[1] = (len(data) + 4) & 0xFF
-            # Add 4 bytes of headers to match RadioHead library.
-            # Just use the defaults for global broadcast to all receivers
-            # for now.
-            self._BUFFER[2] = tx_header[0]  # Header: To
-            self._BUFFER[3] = tx_header[1]  # Header: From
-            self._BUFFER[4] = tx_header[2]  # Header: Id
-            self._BUFFER[5] = tx_header[3]  # Header: Flags
-            device.write(self._BUFFER, end=6)
-            # Now send the payload.
-            device.write(data)
+        # Combine header and data to form payload
+        payload = bytearray(4)
+        if destination is None:  # use attribute
+            payload[0] = self.destination
+        else:  # use kwarg
+            payload[0] = destination
+        if node is None:  # use attribute
+            payload[1] = self.node
+        else:  # use kwarg
+            payload[1] = node
+        if identifier is None:  # use attribute
+            payload[2] = self.identifier
+        else:  # use kwarg
+            payload[2] = identifier
+        if flags is None:  # use attribute
+            payload[3] = self.flags
+        else:  # use kwarg
+            payload[3] = flags
+        payload = payload + data
+        # Write payload to transmit fifo
+        self._write_fifo_from(payload)
         # Turn on transmit mode to send out the packet.
         self.transmit()
         # Wait for packet sent interrupt with explicit polling (not ideal but
         # best that can be done right now without interrupts).
         start = time.monotonic()
         timed_out = False
         while not timed_out and not self.packet_sent:
-            if (time.monotonic() - start) >= timeout:
+            if (time.monotonic() - start) >= self.xmit_timeout:
                 timed_out = True
-        # Listen again if necessary and return the result packet.
+        # Listen again if requested.
         if keep_listening:
             self.listen()
-        else:
-            # Enter idle mode to stop receiving other packets.
+        else:  # Enter idle mode to stop receiving other packets.
             self.idle()
-
         return not timed_out
 
+    def send_with_ack(self, data):
+        """Reliable Datagram mode:
+           Send a packet with data and wait for an ACK response.
+           The packet header is automatically generated.
+           If enabled, the packet tranmsiion will be retried on failure
```

Comment:
@jerryneedell Small typo in this comment `transmission`

Reply:
done